server_src/utils/npm.utils.js

```diff
@@ -1,37 +1,39 @@
 const util = require("util");
 const exec = util.promisify(require("child_process").exec);
 
-async function doWeHaveMissingPackages() {
-  const { stdout, stderr } = await exec("npm outdated --production");
+async function returnListOfMissingPackages() {
+  try {
+    const outdatedPackageJsonList = await exec(
+      "npm outdated --production --json"
+    );
+    const outdatedPackageListParsed = JSON.parse(
+      outdatedPackageJsonList.stdout
+    );
+    if (outdatedPackageListParsed) {
+      const keys = Object.keys(outdatedPackageListParsed);
+      const missingPackageList = [];
 
-  if (stderr) {
-    throw `Could not reliably check for outdated npm packages, is npm installed? | ${stderr}`;
-  }
-
-  if (stdout) {
-    const matchMissingPackages = new RegExp("((.*?)MISSING.*)", "g");
-    const missingPackages = stdout.match(matchMissingPackages);
-    if (!!missingPackages) {
-      return missingPackages.length > 0;
-    } else {
-      return false;
+      keys.forEach((key, index) => {
+        if (!outdatedPackageListParsed[key].current) {
+          missingPackageList.push(key);
+        }
+      });
+      return missingPackageList;
     }
+  } catch (e) {
+    throw `Error running npm outdated command | ${e}`;
   }
 }
 
-async function installMissingNpmDependencies() {
-  const { stdout, stderr } = await exec("npm install --production");
-
-  if (stderr) {
-    throw `Could not execute npm install, is npm installed? | ${stderr}`;
-  }
-
-  if (stdout) {
-    return true;
+async function installNpmDependencies(dependency) {
+  try {
+    await exec(`npm install ${dependency}`);
+  } catch (e) {
+    throw `Error running installation command | ${e}`;
```

Comment:
Not too much different just got rid of the stdout / stderr define and left it with a try/catch similar to the rest of the update functions.